pretyping/evarsolve.mli

```diff
@@ -16,9 +16,17 @@ type alias
 
 val of_alias : alias -> EConstr.t
 
-type allowed_evars =
-| AllowAll
-| AllowFun of (Evar.t -> bool)
+module AllowedEvars : sig
+
+  type t
+
+  val all : t
+  val mem : t -> Evar.t -> bool
+  val remove : Evar.t -> t -> t
+  val except : Evar.Set.t -> t
+  val from_pred : (Evar.t -> bool) -> t
```

Comment:
I believe this API is error-prone, because the functions above are negative checks when this one is positive. Maybe tweak the name?

Reply:
I'm not sure I follow: `AllowedEvars.from_pred f` takes a predicate `f` which says whether an evar is allowed or not. How's the naming wrong?

Reply:
`mem` is the only check, and it is positive

Reply:
Dunno, with the `except` and `remove` above I find it confusing whether it's adding or removing stuff at first sight.

Reply:
I'll move it up (with `mem`) and add comments.